### ISSUE TITLE
https://datatracker.ietf.org/doc/draft-ietf-*/history/ contains hyperlinks links to drafts, but these links redirect to the final RFC

### ISSUE THREAD

**Issue by dilyanpalauzov** [ietf-tools/datatracker#7891]
### Describe the issue

I visit https://datatracker.ietf.org/doc/draft-ietf-sieve-managesieve/history/ and there I can click on the hyperlinks:

* draft-ietf-sieve-managesieve-09.txt ↦ https://datatracker.ietf.org/doc/draft-ietf-sieve-managesieve/09/, or
* draft-ietf-sieve-managesieve-06.txt ↦ https://datatracker.ietf.org/doc/draft-ietf-sieve-managesieve/06/

I expect after clicking on the link to read the respective version of the draft. Instead I am always redirected to the final RFC — https://datatracker.ietf.org/doc/rfc5804/.

I want to read the drafts.

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by jennifer-richards** [comment]
For me, the first example you give (https://datatracker.ietf.org/doc/draft-ietf-sieve-managesieve/09/) goes to the expected document. Others do redirect to the RFC. Looking at the list of versions, it seems that not all the revisions are available. It only seems to have the last rev of each draft shown in the history. The others redirect to the RFC.

We seem to have the other versions in the archive so I'm not sure why it's only listing those versions as available.

<img width="135" alt="image" src="https://github.com/user-attachments/assets/7d017b38-d217-4e3b-8ab4-df353a9b2013">

**Comment by rjsparks** [comment]
See #4933. We don't have DocHistory (that is, we don't know things like who the shepherd was) for drafts that existed before the datatracker came into existance. We've been modifying things to show that we have old versions, and plan to change the UX to show all the old versions when we have them (not just those we have DocHistory for) with a disclaimer that we don't have the metadata for that version.

**Comment by dilyanpalauzov** [comment]
> Looking at the list of versions, it seems that not all the revisions are available. 

The same page offers to compare two revisions, e.g. by generating the hyperlink https://author-tools.ietf.org/iddiff?url1=draft-ietf-sieve-managesieve-08&url2=draft-ietf-sieve-managesieve-06&difftype=--html .  There both revisions are available.  So all revisions are available somewhere.

**Comment by rjsparks** [comment]
Yes, we have the bits (for _almost_ all revisions of all drafts) - and the datatracker knows about them via Submission objects (see again the discussion in #4933) - but we don't have the _metadata_ for many of the fields that would show on the front page - hence the planned change I describe above.

**Comment by rjsparks** [comment]
At the moment, you can view at all the known versions using the /html/ views. See https://datatracker.ietf.org/doc/html/draft-ietf-sieve-managesieve-09 for example.

**Comment by dilyanpalauzov** [comment]
Currently https://datatracker.ietf.org/doc/draft-ietf-sieve-managesieve/history/ generates as hyperlink

draft-ietf-sieve-managesieve-07.txt → https://datatracker.ietf.org/doc/draft-ietf-sieve-managesieve/07/

I am asking to change this to the /html/ view:

draft-ietf-sieve-managesieve-07.txt → https://datatracker.ietf.org/doc/html/draft-ietf-sieve-managesieve-07

This will solve the current case perfectly.
